### PR TITLE
move build to gradle, testing to spock and CI to github actions

### PR DESCRIPTION
**Pull request by oalyman**
* Use gradle instead of maven
* Use spock for testing
* Use github action workflow instead of travis CI

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            credentials {
+                username = "${nexus_user}"
+                password = "${nexus_pw}"
+            }
+        }
+    }
+}
+
+dependencies {
+    implementation "org.codehaus.groovy:groovy-all:2.4.12"
```
Comment (metmajer) [inline build.gradle]: @oalyman any specific reasons you're not using the same dependencies, we use in the MRO shared library?

Comment (oalyman) [reply build.gradle]: 2.4.12 is the groovy version jenkins supports.

Comment (metmajer) [reply build.gradle]: @oalyman good finding! Do you want to provide a quick PR for the MRO?

Comment (oalyman) [reply build.gradle]: Sure. I'll do it in a second

**Pull request merged**